src/compositor/seat.rs

```diff
@@ -26,6 +29,49 @@ impl Seat {
                meta: false,
                ..Seat::default() }
     }
+
+    pub fn clear_focus(&mut self) {
+        if let Some(focused_view) = self.focused.take() {
+            focused_view.activate(false);
+        }
+        with_handles!([(seat: {&mut self.seat})] => {
+            seat.keyboard_clear_focus();
+        }).unwrap();
+    }
+
+    pub fn focus_view(&mut self, view: Rc<View>, views: &mut Vec<Rc<View>>) {
+        if let Some(ref focused) = self.focused {
+            if focused == &view {
+                return
+            }
+            focused.activate(false);
+        }
+        self.focused = Some(view.clone());
+        view.activate(true);
+
+        if let Some(idx) = views.iter().position(|v| *v == view) {
+            let v = views.remove(idx);
+            views.insert(0, v);
```

Comment:
If we are going with this strategy, we should switch to a [linked list](https://doc.rust-lang.org/std/collections/struct.LinkedList.html). 

Note I have **not** profiled the two approaches, I'm just going with my gut here. It's possible this will decrease performance.

We can leave this off for another patch.

Reply:
Unscientific data point: Awesome's C code does not have any linked lists and uses a homegrown vector / array list for all its lists (stacking order, 'visible' order, configured key bindings, ...). This only ever was a problem with "stupid" code that repeatedly scanned the list from the beginning and caused quadratic behaviour that way.

Reply:
what about a VecDeque?

Reply:
Sounds good to me.

Reply:
Just found out that you can't slice a VecDeque though so I don't think you can take slices for those pointer methods.

Reply:
https://doc.rust-lang.org/std/collections/struct.VecDeque.html#method.as_slices

Reply:
Ah but you might need to combine the slices....

eh just leave this for now. It's fine.